enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+    - `Success`
+* `platform` - currently we deploy to 8 platforms: `aws`, `azure`, `gcp,` `libvirt`, `none`, `openstack`, `ovirt`, and `vsphere`. The list will grow. 
+* `os` - You can find all the available combinations of [Go OS and arch values](https://github.com/golang/go/blob/master/src/go/build/syslist.go) with `go tool dist list`. At this stage I assume we only want to colllect operating systems and some of the values are unlikely/impossible:
+  - `linux`
+  - `darwin` - this could be rewritten to `macos` on the installer client if we prefer
+  - `bsd` - this would encompass the `freebsd`, `netbsd`, and `openbsd` values
+  - `windows` - future proofing? 
+  - `other` - everything else not captured above
+* `version` - represents minor releases; assuming we do not backport metrics, `4.5` will be the only value if we make it into the next release and we could expect roughly 20 values over the next five years.
+* `le` - the standard Prometheus label for histogram buckets, this represents how long a command took complete. The current proposal is to have a 5 minute step starting at 15 minutes and maxing at 60.
+
+##### cluster_installation_waitfor
+UPI installs utilize the wait-for command, which would be tracked with a similar histogram:
+```
+# HELP cluster_installation_waitfor represents a single run of wait-for by the OpenShift installer.
+# TYPE cluster_installation_waitfor histogram
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="5"} 0
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="10"} 0
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="15"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="20"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="25"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="30"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="+Inf"} 1
+cluster_installation_waitfor_sum{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux"} 15
+cluster_installation_waitfor_count{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux"} 1
+```
+
+###### Label Values
+Potential label values are the same as the previous metric, except:
+* `le` has values between 5 and 30 minutes
+* `target` and `result` are dependent:
+  - `bootstrap-complete`
+      - `BootstrapFailed`
+      - `APIFailed`
+      - `Success`
+  - `install-complete`
+      - `OperatorsDegraded`
+      - `Success`
+
+##### cluster_installation_manifests and cluster_installation_ignition
```

Comment:
@abhinavdahiya is unclear to me whether we want to collect metrics for every invocation of `create install-config`, but if so I can add a simlar metric to this